spec/models/service_spec.rb

```diff
@@ -65,6 +66,155 @@
       end
     end
 
+    context 'with a user generated slug' do
+      describe 'with invalid slug' do
+        context 'when the slug contains capital letters' do
+          before do
+            subject.slug = 'aCaptialLetter'
+          end
+
+          it 'adds an error on slug' do
+            subject.valid?
```

Comment:
I don't think you need the `subject.slug = 'aCapitalLetter' in a before block for every example - it just adds verbosity.  We usually do a before block when the same code has to be executed before each example, but here it's different every time.  So get rid of the before block, and just assign the slug immediately before checking the validity.

Reply:
Made the suggested changes